lib/ecto/changeset.ex

```diff
@@ -528,6 +528,11 @@ defmodule Ecto.Changeset do
         {changes, errors, valid?}
       :invalid ->
         {changes, [{key, {"is invalid", [type: type, validation: :cast]}} | errors], false}
+      {:invalid, custom_errors} ->
+        {msg, custom_errors} = Keyword.pop(custom_errors, :message)
+        msg = msg || "is invalid"
+        new_errors = Keyword.merge([type: type, validation: :cast], custom_errors)
```

Comment:
Should Ecto raise an error if the custom error contains reserved keys like `:type` and `:validation`?

Reply:
I think we want to allow the user to change `validation` but I think is no reason to allow the type to be changed. I would go with something like:

```elixir
[type: type] ++ Keyword.put_new(custom_errors, :validation, :cast)
```

Reply:
I've changed it to the remarks of @josevalim. If we want to inform the user that we are overriding the `type`, that's something for the docs?